crates/ruff_linter/src/rules/flake8_comprehensions/rules/unnecessary_dict_comprehension_for_iterable.rs

```diff
@@ -96,6 +96,14 @@ pub(crate) fn unnecessary_dict_comprehension_for_iterable(
         return;
     }
 
+    // Don't suggest `dict.fromkeys` if the target is an attribute, subscript, or slice (side-effecting assignment).
+    match &generator.target {
+        Expr::Attribute(_) | Expr::Subscript(_) | Expr::Slice(_) => {
+            return;
+        }
+        _ => {}
+    }
```

Comment:
With only one `match` arm, I think it's slightly more idiomatic to use `matches!`:


```suggestion
    if matches!(generator.target, Expr::Attribute(_) | Expr::Subscript(_) | Expr::Slice(_)) {
            return;
    }
```